### ISSUE TITLE
python full version in pyproject.toml causes compatibility issues in requirements.txt

### ISSUE THREAD

**Issue by ergleb78** [python-poetry/poetry#6804]
- **Poetry version**: 1.2.2
- **Python version**: 3.8.10
- **OS version and name**: Ubuntu 22.04 
- **pyproject.toml**: python = ">=3.10,<3.11"

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [ X] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [X ] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [X ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue
I'm not sure if it's a bug or even poetry's bug. 
The relevant section of pyproject.toml:

[tool.poetry.dependencies]
python = "3.10.5"

I'm using poetry with Bazel. To do so, I have to export dependencies to requirements.txt using the following command:
poetry export -f requirements.txt -o requirements.txt --without-hashes --with dev

If I specify version like above my requirements.txt looks like this:
....
deepdiff6==6.2.0 ; python_full_version == "3.10.5"
....

Bazel python rules seem to expect python_version to be present and look similar to
....
click==8.1.3 ; python_version >= "3.10" and python_version < "3.11"
....
so the export above breaks the bazel's workspace repo import mechanism. 

I found a workaround, to define python version like this, so poetry includes python_version in requirements.txt
[tool.poetry.dependencies]
python = ">=3.10,<3.11"

Is this expected behaviour?

**Comment by dimbleby** [comment]
yes this is excpected behaviour.  The export is supposed to match the input.
